cli/cmd/plugin/conformance/main.go

```diff
@@ -4,29 +4,37 @@
 package main
 
 import (
+	"log"
 	"os"
 
 	conformance "github.com/vmware-tanzu/community-edition/cli/cmd/plugin/conformance/pkg"
 
-	klog "k8s.io/klog/v2"
-
 	cliv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cli/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/plugin"
 )
 
 var descriptor = cliv1alpha1.PluginDescriptor{
 	Name:        "conformance",
 	Description: "Run Sonobuoy conformance tests against clusters",
 	Group:       cliv1alpha1.RunCmdGroup,
+	Version:     cli.BuildVersion,
 }
 
+var (
+	defaultVersion = "v0.0.1-unversioned"
+)
+
 var logLevel int32
 
 func main() {
+	// if descriptor.Version == "" {
+	// 	descriptor.Version = defaultVersion
+	// }
```

Comment:
I was able to get this to build by uncommenting this chunk and the following chunk:
```golang
if descriptor.Version == "" {
    descriptor.Version = defaultVersion
}
..

c, _, err := p.Cmd.Find([]string{"version"})
...
```

We _do_ need the conformance Version command to replace the default one since it also displays the sonobouy version.

Hmm but when I build with `make release`, I see the following error:
```
+ cp -f hack/release/../../artifacts/conformance/v0.9.2-dev.1/tanzu-conformance-linux_amd64 hack/release/../../build/tce-linux-amd64-v0.9.2-dev.1/bin/tanzu-plugin-conformance
cp: cannot stat 'hack/release/../../artifacts/conformance/v0.9.2-dev.1/tanzu-conformance-linux_amd64': No such file or directory
make: *** [Makefile:228: package-release] Error 1
```
and I notice that there are only binaries with the "unversioned" version:
```
❯ ll artifacts/conformance
total 8.0K
-rw-r--r-- 1 jmcb jmcb  227 Nov 11 10:51 plugin.yaml
drwxrwxr-x 3 jmcb jmcb 4.0K Nov 11 10:51 v0.0.1-unversioned
```

I wonder if maybe something changed with the LdFlags and we're picking up the `defaultVersion` automatically and not our passed in version ... 

Maybe @nrb has thoughts here too

Reply:
Looks like pkg/v1/cli.BuildVersion was deprecated in favor of pkg/v1/buildInfo.Version

vmware-tanzu/tanzu-framework@3162254#diff-287b2c13779f4ba2d8e41cc7c201cb67605ab8ef8cce54a977df7a5c83b00dddR20-R21

Reply:
FYI - I was able to get this to work here: https://github.com/vmware-tanzu/community-edition/pull/2523